src/main/java/seedu/duke/models/schema/Timetable.java

```diff
@@ -0,0 +1,195 @@
+            }
+        }
+    }
+
+    /**
+     * Modifies the timetable for the specified student based on user input.
+     * @author @rohitcube
+     * @param student The student object.
+     * @throws seedu.duke.exceptions.InvalidModifyArgumentException If an invalid argument is provided.
+     */
+    public void modifyTimetable(Student student) throws seedu.duke.exceptions.InvalidModifyArgumentException {
+        try {
+            System.out.println("List of modules in current semester: ");
+            if (currentSemesterModulesWeekly.isEmpty()) {
+                System.out.println("There are no modules in your current semester. " +
+                        "Please add in modules, or generate using the 'recommend' command.");
+                return;
+            }
+            for (int i = 0; i < currentSemesterModulesWeekly.size(); i++) {
+                System.out.println(currentSemesterModulesWeekly.get(i).getModuleCode());
+            }
+            Scanner in = new Scanner(System.in);
+            System.out.println("Which current module do you want to modify? (ENTER MODULE CODE)");
+            String moduleCode = in.nextLine().trim();
+            if (!isExistInCurrentSemesterModules(moduleCode, timetable.currentSemesterModulesWeekly)) {
+                System.out.println("Sorry that module doesn't exist in current semesters");
+                return;
+            }
+            System.out.println("Ok that module exists. Enter what you would like to change in this way " +
+                    "(lecture, tutorial, lab):\n " +
+                    "[lecture /time 12 /duration 3 /day Tuesday], time range of values: 8-20");
+            String userInput = in.nextLine().trim();
+            // pass in the ModuleWeekly element from currentSemester
+            int indexOfModuleWeeklyToModify = getIndexOfModuleWeekly(moduleCode, currentSemesterModulesWeekly);
+            processModifyArguments(userInput, indexOfModuleWeeklyToModify, student);
+        } catch (seedu.duke.exceptions.InvalidModifyArgumentException e) {
+            throw new seedu.duke.exceptions.InvalidModifyArgumentException();
+        }
+    }
+
+    /**
+     * Processes the modify arguments provided by the user and updates the module schedule accordingly.
+     * @author @rohitcube
+     * @param userInput     The user input specifying the modification.
+     * @param indexOfModule The index of the ModuleWeekly object to be modified.
+     * @param student       The student object.
+     * @throws seedu.duke.exceptions.InvalidModifyArgumentException If an invalid argument is provided.
+     */
+    public void processModifyArguments(String userInput, int indexOfModule, Student student)
+            throws seedu.duke.exceptions.InvalidModifyArgumentException {
+        try {
+            int startIndexOfStart = userInput.indexOf("/time");
+            String command = userInput.substring(0, startIndexOfStart).trim().toUpperCase();
```

Comment:
By parsing the command using indexOf('/time'), if the user inputs a command without a /time, it would bypass the first guard clause. Which renders the guard clause useless
```
if (!command.equals("LECTURE") &&
                     !command.equals("TUTORIAL") &&
                     !command.equals("LAB")) {
                 System.out.println("Not a valid command. Please try again!");
                 return;
             }
```

Recommended way to do it 

```
userInput = in.nextLine().trim().replace("\r", "");
String[] wordArray = userInput.split(" ");
String command = wordArray[0].toUpperCase();
```